core/src/main/java/org/mule/runtime/core/internal/construct/AbstractPipeline.java

```diff
@@ -353,6 +362,10 @@ boolean isCompleteSignalRejectedExecutionException(Throwable throwable) {
   }
 
   private void fireCompleteNotification(CoreEvent event, MessagingException messagingException) {
+    FlowCallStack flowCallStack = event.getFlowCallStack();
```

Comment:
Shouldn't the push and pop methods be part of the interface? This will always be a stack.

Reply:
same as before